cve_bin_tool/format_checkers.py

```diff
@@ -80,6 +80,37 @@ def update_checker_table(file_path, markdown):
         f.writelines(lines)
 
 
+def update_number_of_checkers(file_path,number_of_checkers):
+    """Updates README.md with the new number of checkers"""
+    lines = []
+    number_of_checkers=str(number_of_checkers)
+    # using -1 for unbound indices
+    start_index: int = -1
+
+    with open("README.md") as f:
+            for index, line in enumerate(f):
+                if "NUMBER OF CHECKERS START" in line:
+                    start_index = index
+                lines.append(line)
+
+    words=lines[start_index].split(">")
+    words[0]=words[0]+">"
+    words[1]=words[1]+">"
+    sublist_words=words[1].split("<")
+    sublist_words[1]="<"+sublist_words[1]
+    words.pop(1)
+    words.insert(1,sublist_words[1])
+    words.insert(1,number_of_checkers)
+    new_line=""
+    for word_txt in words:
+        new_line+=word_txt
+
+    lines = lines[: start_index] + [new_line] + lines[start_index+1:]
+
+    with open(file_path, "w") as f:
+        f.writelines(lines)
+
+
```

Comment:
I'm curious if something like `re.sub('<!-- NUMBER OF CHECKERS START-->.*<!-- NUMBER OF CHECKERS END-->', number_of_checkers, lines, flags=re.DOTALL)` (might need some more refining) can work.

Basically replace the number between the 2 comments (markers) with the current no. of checkers.

Reply:
The current solution is also working, though I am not sure why the linter error are arising.

Reply:
However, I can try out the above mentioned method if required.

Reply:
> The current solution is also working, though I am not sure why the linter error are arising.

It seems like you need to run black on the file for formatting it. If you haven't setup pre-commit and it's hooks take a look at https://cve-bin-tool.readthedocs.io/en/latest/CONTRIBUTING.html#running-linters. And then black, isort etc. will run automatically while committing.

> However, I can try out the above mentioned method if required.

This would be awesome. Using regex would increase the readability of the code a lot more.

Reply:
> > The current solution is also working, though I am not sure why the linter error are arising.
> 
> It seems like you need to run black on the file for formatting it. If you haven't setup pre-commit and it's hooks take a look at https://cve-bin-tool.readthedocs.io/en/latest/CONTRIBUTING.html#running-linters. And then black, isort etc. will run automatically while committing.
> 
> > However, I can try out the above mentioned method if required.
> 
> This would be awesome. Using regex would increase the readability of the code a lot more.

Thanks for the help

Reply:
Thanks for the help on the linting